scene/main/viewport.cpp

```diff
@@ -343,7 +343,13 @@ void Viewport::_sub_window_update(Window *p_window) {
 		close_icon->draw(sw.canvas_item, r.position + Vector2(r.size.width - close_h_ofs, -close_v_ofs));
 	}
 
-	RS::get_singleton()->canvas_item_add_texture_rect(sw.canvas_item, r, sw.window->get_texture()->get_rid());
+	Transform2D xform = sw.window->window_transform * sw.window->stretch_transform;
+	Rect2i vr = xform.xform(sw.window->get_visible_rect());
+	vr.position += p_window->get_position();
+	if (vr != r) {
+		RS::get_singleton()->canvas_item_add_rect(sw.canvas_item, r, Color());
```

Comment:
```suggestion
		const Color clear_color = RS::get_singleton()->get_default_clear_color();
		RS::get_singleton()->canvas_item_add_rect(sw.canvas_item, r, clear_color);
```
This way it's consistent with the other modes:
(clear color changed via project settings)

![image](https://github.com/user-attachments/assets/783ea0d1-4678-4fe7-923f-152a234bef3c)

Reply:
When changing the clear color and viewing the windows in a not-embedded form, then there are black bars, that aren't affected by the clear color.

![image](https://github.com/user-attachments/assets/52969431-2264-409c-9558-938f60f89866)

My understanding is:
- The black color indicates, that in this area, there is no `Viewport`-node
- The clear color indicates, that in this area, there is the background of a `Viewport`-node

So the distinction between black color and clear color helps to visualize, where the `Viewport` ends.

This distinction between Viewport-area and non-Viewport-area would vanish, if these bars would also be set to clear color.

I would prefer to not follow your suggestion, so that this visual distinction is kept for the user and that the display is kept consistent between embedded and non-embedded windows.